src/coreclr/nativeaot/BuildIntegration/Microsoft.NETCore.Native.Unix.targets

```diff
@@ -164,6 +164,11 @@ The .NET Foundation licenses this file to you under the MIT license.
       <NativeSystemLibrary Include="crypto" />
     </ItemGroup>
 
+    <ItemGroup Condition="'$(UseSystemBrotli)' != 'false' and Exists('$(IlcSdkPath)nonportable.txt')">
```

Comment:
Can we extend this with something that is similar to https://github.com/dotnet/runtime/pull/107166/files#r1739122333? Then source-build can be built with bundled or system brotli.

Otherwise we need to add `--cmakeargs -DCLR_CMAKE_USE_SYSTEM_BROTLI=true` (similar to what is in the diff in https://github.com/dotnet/runtime/issues/107020#issuecomment-2320204908) and source-building requires system brotli.

Reply:
For .NET 10, we will get it with #107166.

For .NET 9, it would require build refactoring that is hard to justify for a backport at this point.

Reply:
ok, then we need to add something that sets `CLR_CMAKE_USE_SYSTEM_BROTLI=true` so maintainers provide brotli in their non-portable builds.

Without it, the brotli the bootstrap build will work (since Microsoft's portable SDK doesn't have a `nonportable.txt` breadcrumb) but successive builds using the produced non-portable SDK won't, because NativeAOT will fail when trying to find brotli on a system that doesn't provide it.

Reply:
Why is this not sufficient https://github.com/dotnet/runtime/blob/main/eng/DotNetBuild.props#L91 ?

Reply:
That doesn't do anything unless the maintainer builds the vmr with `--with-system-libs brotli`.

If he doesn't add it, his bootstrap build works, but he is not aware that successive builds using that SDK will fail unless he provides brotli.

By adding `CLR_CMAKE_USE_SYSTEM_BROTLI=true` for non-portable, we make this issue visible also on the bootstrap build.

Reply:
> That doesn't do anything unless the maintainer builds the vmr with --with-system-libs brotli.

Are there any distros that do not want to use `--with-system-libs brotli`? I think that the conversation in https://github.com/dotnet/runtime/issues/107020 suggested that all interested distro builds want to use `--with-system-libs brotli`.

Reply:
Yes, there are no maintainers that have objected to using system brotli, and we are assuming brotli to be a common available system library.

Note that this change will also require a brotli development package to be added to the source-build CI jobs.

Reply:
By that you mean added to the container?

Reply:
Yes.